### PR TITLE
Retry all Spnego authentication on network issue

### PR DESCRIPTION
**Pull request by kokosing**
Retry all Spnego authentication on network issue

### CODE REVIEW

File: http-client/src/main/java/io/airlift/http/client/RetryDriver.java [hunk 1]
```diff
@@ -122,6 +122,14 @@ public RetryDriver stopOnIllegalExceptions()
         return stopOn(NullPointerException.class, IllegalStateException.class, IllegalArgumentException.class);
     }
 
+    public void run(String callableName, Runnable runnable)
+            throws Exception
```
Comment (findepi) [inline http-client/src/main/java/io/airlift/http/client/RetryDriver.java]: needed?

File: http-client/src/main/java/io/airlift/http/client/spnego/SpnegoAuthentication.java [hunk 2]
```diff
@@ -107,6 +107,21 @@ public URI getURI()
...
+                try {
+                    RetryDriver.retry()
+                            .stopOn(Predicates.not(SpnegoAuthentication::networkIssue))
+                            .run("authenticate", () -> authenticate(request));
+                }
+                catch (RuntimeException e) {
+                    throw e;
+                }
+                catch (Exception e) {
+                    throw new RuntimeException(format("Failed to authenticate request %s", request.getURI()), e);
```
Comment (findepi) [inline http-client/src/main/java/io/airlift/http/client/spnego/SpnegoAuthentication.java]: This additional information might be useful also when `RuntimeException` is thrown.

**Comment by sopel39** [comment]
`release-0.176-e`?

File: http-client/src/main/java/io/airlift/http/client/spnego/SpnegoAuthentication.java (hunk 2 above)
Comment (kokosing) [reply http-client/src/main/java/io/airlift/http/client/spnego/SpnegoAuthentication.java]: We should not reach here. All exception in `authenticate` are wrapped in `RuntimeException`. This `catch` is to make compiler happy.

**Comment by kokosing** [comment]
> Does it help? 😉

:) I do not know.

File: http-client/src/main/java/io/airlift/http/client/RetryDriver.java (hunk 1 above)
Comment (kokosing) [reply http-client/src/main/java/io/airlift/http/client/RetryDriver.java]: To pass a `void method` to `run`

**Pull request merged**